neutron/beamline.cpp

```diff
@@ -58,5 +63,50 @@ Variable two_theta(const Dataset &d) {
 
   return acos(dot(beam, scattered));
 }
+} // namespace beamline_impl
+
+core::VariableConstProxy position(const core::DatasetConstProxy &d) {
```

Comment:
Is it worth templating all of these?
Or maybe this way is better because you prevent them being used for types other than `DatasetConstProxy` and `DataConstProxy`?

Reply:
Correct, I did want to avoid redundant instantiation, instead relying on implicit conversion to the proxy types.